src/protean/core/entity.py

```diff
@@ -355,6 +357,27 @@ def delete(self):
 
         return deleted_item_count
 
+    def update_all(self, *args, **kwargs):
+        """Updates all objects with details given if they match a set of conditions supplied.
+
+        This method forwards filters and updates directly to the adapter. It does not instantiate
+        models and it does not trigger Entity callbacks or validations.
+
+        Update values can be specified either as a dict, or keyword arguments.
+
+        Returns the number of objects matched (which may not be equal to the number of objects
+            updated if objects rows already have the new value).
+        """
+        updated_item_count = 0
+        _, adapter = self._retrieve_model()
+        try:
+            updated_item_count = adapter._update_all_objects(self._criteria, *args, **kwargs)
```

Comment:
all adapter methods like `_update_all_objects` `_update_objects` should be made public as they are accessed publically

Reply:
Good point! 👍 Will rename them to be valid public methods.